consensus/src/zk.rs

```diff
@@ -74,7 +74,7 @@ impl Authority for ZookeeperAuthority {
                 .exists_w(CONTROLLER_KEY, UnparkWatcher::new())
                 .is_ok()
             {
-                thread::park();
+                thread::park_timeout(Duration::from_secs(60));
```

Comment:
You may want this timeout to be shorter. If this race happens, we don't really want to wait a minute for things to recover

Reply:
We're balancing how quickly we respond to a *really* unlikely race with the frequency at which we needlessly poll Zookeeper if the race doesn't happen